src/multiselect_contextmenu.js

```diff
@@ -547,20 +547,18 @@ const registerDelete = function() {
   const deleteOption = {
     displayText: function(scope) {
       let descendantCount = 0;
-      const workspace = scope.block.workspace;
-      const blockSelection = blockSelectionWeakMap.get(workspace);
-      blockSelection.forEach(function(id) {
-        const block = workspace.getBlockById(id);
-        if (block && !hasSelectedParent(block)) {
-          // Count the number of blocks that are nested in this block.
+      const countDescendants = function(block) {
+        if (block) {
           descendantCount += block.getDescendants(false).length;
           const nextBlock = block.getNextBlock();
           if (nextBlock) {
-            // Blocks in the current stack would survive this block's deletion.
             descendantCount -= nextBlock.getDescendants(false).length;
           }
```

Comment:
Hmm... we probably need to rethink this logic at some point. It seems wasteful in large projects to count all of the descendants of the current block + its next block only to count all of the descendants of the next block and subtract them. In large block stacks I imagine this performs poorly. I know that's technically not relevant to this particular fix but it's something to think about.

Reply:
I agree. We are also returning large arrays of blocks when we only need the count, but that seems to be the only API available.

An alternative could be get id of nextBlock if it exists, use getChildren directly, discard next block by id, then loop over recursively like in getDescendants and then return the count. Feels rather convoluted and I am not sure how reliable the ids are. Thoughts?